BlotoutAnalytics/BlotoutAnalytics.m

```diff
@@ -548,12 +548,72 @@ -(void)checkAppTrackingStatus {
         }
       }
       
-      BOACaptureModel *model = [[BOACaptureModel alloc] initWithEvent:@"App Tracking" properties:@{@"status":statusString,@"idfa":idfaString} eventCode:@(BO_APP_TRACKING) screenName:nil withType:BO_SYSTEM];
+      BOACaptureModel *model = [[BOACaptureModel alloc] initWithEvent:@"App Tracking" properties:@{@"status":statusString,@"idfa":idfaString}  screenName:nil withType:BO_SYSTEM];
       [self.eventManager capture:model];
     }];
   } @catch(NSException *exception) {
     BOFLogDebug(@"%@", exception);
   }
 }
 
+-(void)captureTransaction:(nonnull TransactionData*)transactionData withInformation:(nullable NSDictionary*)eventInfo {
+  @try {
+    if (!self.isEnabled) {
+      return;
+    }
+    
+    [[BOEventsOperationExecutor sharedInstance] dispatchEventsInBackground:^{
+      NSMutableDictionary *transactionInfo = [[NSMutableDictionary alloc] initWithObjects:[NSArray arrayWithObjects:transactionData.transaction_id, transactionData.transaction_currency,transactionData.transaction_total,transactionData.transaction_discount,transactionData.transaction_shipping,transactionData.transaction_tax,nil] forKeys:[NSArray arrayWithObjects:@"transaction_id", @"transaction_currency",@"transaction_total",@"transaction_discount",@"transaction_shipping",@"transaction_tax",nil]];
```

Comment:
why do we need nil at the end?

Reply:
That is part of the syntax, nil termination is required. It wouldn't compile otherwise giving a "missing sentinel" error.